### PR TITLE
fix(crosswalk_traffic_light_estimator): move crosswalk after fusion

### PR DESCRIPTION
**Pull request by tzhong518**
### <samp>🤖 Generated by Copilot at b5458c5</samp>

This pull request refactors the traffic light recognition and estimation modules to use the new `TrafficSignal` messages and improve modularity. It simplifies the `traffic_light_node_container.launch.py` file, updates the topic names and launch arguments in the `traffic_light.launch.xml` file, and adds a crosswalk traffic signal estimation feature. It also updates the `crosswalk_traffic_light_estimator` node and package to use the `autoware_perception_msgs` package and the `traffic_signal_id` field.

This pull request improves the traffic light recognition system by simplifying the launch files, updating the topic names and arguments, and replacing the deprecated `tier4_perception_msgs` package with the `autoware_perception_msgs` package. It also adds a new optional node for crosswalk traffic light estimation.

## Related links

[TIER IV INTERNAL LINK](https://tier4.atlassian.net/browse/RT1-3489)
## Tests performed

https://tier4.atlassian.net/wiki/spaces/~459732634/pages/2865332600/traffic+light+bug+fix#Result

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by miursh — APPROVED** [review]
LGTM
jsut one minor comment

File: perception/crosswalk_traffic_light_estimator/src/node.cpp [hunk 1]
```diff
@@ -237,7 +237,7 @@ void CrosswalkTrafficLightEstimatorNode::setCrosswalkTrafficSignal(
       output_traffic_light.color = color;
       output_traffic_light.confidence = 1.0;
       output_traffic_signal.elements.push_back(output_traffic_light);
-      output_traffic_signal.traffic_light_id = ll_traffic_light.id();
+      output_traffic_signal.traffic_signal_id = ll_traffic_light.id();
```
Comment (TomohitoAndo) [inline perception/crosswalk_traffic_light_estimator/src/node.cpp:240]: `traffic_signal_id` should represent the regulatory element id of the traffic light.
Therefore this node needs to do the [groupFusion](https://github.com/tier4/autoware.universe/blob/ed5fdace2cc26dd1c353c9daf6aa738ca4f6293c/perception/traffic_light_multi_camera_fusion/src/node.cpp#L292-L321), which is currently performed by `traffic_light_multi_camera_fusion`.

Current Message Output:
```
signals:
- traffic_signal_id: 144907
  elements:
  - color: 0
    shape: 0
    status: 0
    confidence: 1.0
- traffic_signal_id: 144918
  elements:
  - color: 0
    shape: 0
    status: 0
    confidence: 1.0
```

Expected Message Output:
(Note: ID 144912 represents the regulatory element ID which associates with the traffic light IDs 144907 and 144918.)
```
- traffic_signal_id: 144912
  elements:
  - color: 0
    shape: 0
    status: 0
    confidence: 1.0
```

Comment (miursh) [reply perception/crosswalk_traffic_light_estimator/src/node.cpp:240]: Definitely! Thank you for your comment. I apologize for my mistake.

Comment (miursh) [reply perception/crosswalk_traffic_light_estimator/src/node.cpp]: @TomohitoAndo Fixed at b5458c59e18519e71b547068e0fb8dde27bc3a3a

**Comment by miursh** [comment]
Tested with odaiba data.
image
image

**Comment by miursh** [comment]
@TomohitoAndo Should they be merged? I'm not sure for the intended behavior.
This occurs because of the another crosswalk lanelets connected the the same regulatory element id.

**Comment by TomohitoAndo** [comment]
@miursh 
Planning module maybe works well even if there are duplicated data.
But I think duplicated data in the message is confusing and might cause unexpected behavior in the future, so it would be better to be merged. (I think it is okay to fix this in another PR)
What do you think? :pray:

**Comment by miursh** [comment]
@TomohitoAndo Thanks! As we talked, could you make a separate pull request to fix this?

**Comment by miursh** [comment]
@satoshi-ota Could you check this PR since you are the only codeowner.

**Review by TomohitoAndo — APPROVED** [review]

**Review by satoshi-ota — APPROVED** [review]

**Pull request merged**